src/lib/rnp.cpp

```diff
@@ -4079,10 +4076,23 @@ try {
         return ret;
     }
 
+    rnp_output_t output = external_output;
```

Comment:
As we are moving to C++ internals, we should use here ArmoredDest instead of C `rnp_output_t`, as this ensures deallocation handling due to exceptions and so on. Please see how this is implemented in the function `rnp_key_export_autocrypt()`.

Reply:
Addressed.